src/mds/Beacon.cc

```diff
@@ -304,9 +304,9 @@ void Beacon::notify_health(MDSRank const *mds)
   }
 
   // Detect MDS_HEALTH_TRIM condition
-  // Arbitrary factor of 2, indicates MDS is not trimming promptly
+  // Indicates MDS is not trimming promptly
   {
-    if (mds->mdlog->get_num_segments() > (size_t)(g_conf()->mds_log_max_segments * 2)) {
+    if (mds->mdlog->get_num_segments() > (size_t)(g_conf()->mds_log_max_segments * g_conf()->mds_log_warn_factor)) {
```

Comment:
This `g_conf()->mds_log_warn_factor` access only works if the option exists in legacy_options.h. This was for backwards compatibility with the option infrastructure was updated.

You need to instead use the new access method: ` g_conf().get_val<double>("mds_log_warn_factor")`. Then it should compile.

Whoever told you to modify `legacy_options.h` was wrong, sorry.